clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/TransactionResourceDefinition.java

```diff
@@ -244,7 +279,13 @@ public ManagementResourceRegistration register(ManagementResourceRegistration pa
         ManagementResourceRegistration registration = parent.registerSubModel(this);
         parent.registerAlias(LEGACY_PATH, new SimpleAliasEntry(registration));
 
-        ResourceDescriptor descriptor = new ResourceDescriptor(this.getResourceDescriptionResolver()).addAttributes(Attribute.class);
+        Capability dependentCapability = new BinaryRequirementCapability(InfinispanCacheRequirement.CACHE, BinaryCapabilityNameResolver.GRANDPARENT_PARENT);
+        ResourceDescriptor descriptor = new ResourceDescriptor(this.getResourceDescriptionResolver())
+                .addAttributes(Attribute.class)
+                // Add a requirement on the tm capability to the parent cache capability
+                .addResourceCapabilityReference(new TransactionResourceCapabilityReference(dependentCapability, TransactionRequirement.TRANSACTION_MANAGER, EnumSet.of(TransactionMode.NONE, TransactionMode.BATCH)))
+                // Add a requirement on the XAResourceRecoveryRegistry capability to the parent cache capability
+                .addResourceCapabilityReference(new TransactionResourceCapabilityReference(dependentCapability, TransactionRequirement.XA_RESOURCE_RECOVERY_REGISTRY, EnumSet.complementOf(EnumSet.of(TransactionMode.NON_XA))));
```

Comment:
For the XAResourceRecovery capability, the excludedModes should be: EnumSet.complementOf(EnumSet.of(TransactionMode.FULL_XA)), right?

Reply:
@pferraro What you say certainly sounds right!  But I derived this from the following, so if my change is wrong the following probably is too:

https://github.com/wildfly/wildfly/blob/master/clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/TransactionServiceConfigurator.java#L81

Reply:
That link references the TransactionSynchronizationRegistry, which is only required for NON_XA mode.  Does a capability exist for this too?
This is distinct from the XAResourceRecovery, which required for tx durability, but only when using FULL_XA mode.

Reply:
Thanks @pferraro.  In the commit where I added TSR service name lookup at https://github.com/wildfly/wildfly/pull/11745/files#diff-3f6e8c6f5c7154285a4fccd0dd06b25bR99 I forgot to record a capability requirement. And then I confused myself that the TransactionServiceConfigurator thing about NON_XA was for XAResourceRecovery.  I'll correct this.